### PR TITLE
feat(general): broadcast when CRUD method fails

### LINKED ISSUE #4: Feature request publish custom events on unauthorized requests
Because we are using `angular-sails-socket` we don't have the facility to add our own custom interceptor,  that we would otherwise have if we are using plain `$http` 

Problem is we can't handle globally `403` or `401` unauthorized requests

Feature request to publish custom fail events on `$rootScope`, so from my app I could do `on('sails-auth-fail)` then redirect the user to a login screen or something

### PR DESCRIPTION
**Pull request by nadeemkhedr**
Closes #4

### CODE REVIEW

**Comment by nadeemkhedr** [comment]
The Failing is not related to the PR, it should been failing before

File: src/sailsSocket.js [hunk 1]
```diff
@@ -70,14 +70,19 @@ angular.module('bethel.sailsSocket', []).provider('sailsSocket', function() {
       return found;
     }
 
+    function handleError(rejectPromise, response, data) {
+      $rootScope.$broadcast('sailsSocket:error', response, data);
```
Comment (Ignigena) [inline src/sailsSocket.js:90]: For consistency can we have the broadcast order the parameters with `data` first and `response` second?

**Review by Ignigena — CHANGES_REQUESTED** [review]
This looks great! I just fixed the existing failing tests  and it looks like there are a couple ESLint issues in this PR that should be addressed before merging.

File: src/sailsSocket.js [hunk 2]
```diff
@@ -190,32 +200,32 @@ angular.module('bethel.sailsSocket', []).provider('sailsSocket', function() {
...
-          case 'destroyed':
-            const deleteIndex = findIndexById(scope, message.id);
-            if (deleteIndex !== null) {
-              scope.splice(deleteIndex, 1);
+            case 'created':
+              if (findIndexById(scope, message.id) !== null) return;
+              scope.unshift(message.data);
+              break;
+
+            case 'destroyed': {
```
Comment (nadeemkhedr) [inline src/sailsSocket.js:208]: This is actually interesting, didn't know about it http://eslint.org/docs/rules/no-case-declarations

**Comment by nadeemkhedr** [comment]
Done! updated some eslint config and fixed all lint issues

**Comment by nadeemkhedr** [comment]
don't know why travis is failing, looks like it is looking at old code, fixed the lint issues there

**Review by Ignigena — APPROVED** [review]

**Comment by Ignigena** [comment]
Thanks Nadeem 👍

**Pull request merged**